x-pack/plugins/cloud_security_posture/server/create_indices/create_indices.ts

```diff
@@ -4,72 +4,153 @@
-import { latestFindingsMapping } from './latest_findings_mapping';
+import { errors } from '@elastic/elasticsearch';
 import {
-  LATEST_FINDINGS_INDEX_DEFAULT_NS,
-  LATEST_FINDINGS_INDEX_NAME,
   BENCHMARK_SCORE_INDEX_DEFAULT_NS,
-  BENCHMARK_SCORE_INDEX_NAME,
+  BENCHMARK_SCORE_INDEX_PATTERN,
+  BENCHMARK_SCORE_INDEX_TEMPLATE_NAME,
+  CLOUD_SECURITY_POSTURE_PACKAGE_NAME,
   CSP_INGEST_TIMESTAMP_PIPELINE,
+  FINDINGS_INDEX_NAME,
+  LATEST_FINDINGS_INDEX_DEFAULT_NS,
+  LATEST_FINDINGS_INDEX_PATTERN,
+  LATEST_FINDINGS_INDEX_TEMPLATE_NAME,
 } from '../../common/constants';
 import { createPipelineIfNotExists } from './create_processor';
+import { benchmarkScoreMapping } from './benchmark_score_mapping';
 
 // TODO: Add integration tests
+
 export const initializeCspIndices = async (esClient: ElasticsearchClient, logger: Logger) => {
   await createPipelineIfNotExists(esClient, CSP_INGEST_TIMESTAMP_PIPELINE, logger);
+
   return Promise.all([
-    createIndexIfNotExists(
-      esClient,
-      LATEST_FINDINGS_INDEX_NAME,
-      LATEST_FINDINGS_INDEX_DEFAULT_NS,
-      latestFindingsMapping,
-      {},
-      logger
-    ),
-    createIndexIfNotExists(
-      esClient,
-      BENCHMARK_SCORE_INDEX_NAME,
-      BENCHMARK_SCORE_INDEX_DEFAULT_NS,
-      benchmarkScoreMapping,
-      { default_pipeline: CSP_INGEST_TIMESTAMP_PIPELINE },
-      logger
-    ),
+    createLatestFindingsIndex(esClient, logger),
+    createBenchmarkScoreIndex(esClient, logger),
   ]);
 };
 
-export const createIndexIfNotExists = async (
+const createBenchmarkScoreIndex = async (esClient: ElasticsearchClient, logger: Logger) => {
+  try {
+    const INDEX_TEMPLATE_V830 = 'cloud_security_posture.scores';
+    await deleteIndexTemplateSafe(esClient, logger, INDEX_TEMPLATE_V830);
+
+    // We always want to keep the index template updated
+    await esClient.indices.putIndexTemplate({
+      name: BENCHMARK_SCORE_INDEX_TEMPLATE_NAME,
+      index_patterns: BENCHMARK_SCORE_INDEX_PATTERN,
+      template: {
+        mappings: benchmarkScoreMapping,
+        settings: {
+          default_pipeline: CSP_INGEST_TIMESTAMP_PIPELINE,
+          lifecycle: {
+            // This is the default lifecycle name, it is named on the data-stream type (e.g, logs/ metrics)
+            name: 'logs',
+          },
+        },
+      },
+      _meta: {
+        package: {
+          name: CLOUD_SECURITY_POSTURE_PACKAGE_NAME,
+        },
+        managed_by: 'cloud_security_posture',
+        managed: true,
+      },
+      priority: 500,
+    });
+
+    await createIndexSafe(esClient, logger, BENCHMARK_SCORE_INDEX_DEFAULT_NS);
+  } catch (e) {
+    logger.error(
+      `Failed to upsert index template [Template: ${BENCHMARK_SCORE_INDEX_TEMPLATE_NAME}]`
+    );
+    logger.error(e);
+  }
+};
+
+const createLatestFindingsIndex = async (esClient: ElasticsearchClient, logger: Logger) => {
+  try {
+    const INDEX_TEMPLATE_V830 = 'cloud_security_posture.findings_latest';
```

Comment:
should this be `LATEST_FINDINGS_INDEX_NAME` ?

Reply:
no,
the index template name is related to obsolete implementation (`INDEX_TEMPLATE_V830`), using a constant that is relevant to the current implementation (`LATEST_FINDINGS_INDEX_NAME`) would couple them. Which is not desired.

For example, if for any reason the value of `LATEST_FINDINGS_INDEX_NAME` changes we won't delete the old implementation anymore. Therefore I introduced a new constant